src/main/java/com/example/bookstoreapplication/service/category/CategoryServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.example.bookstoreapplication.service.category;
+
+import com.example.bookstoreapplication.dto.category.CategoryDto;
+import com.example.bookstoreapplication.dto.category.CategoryRequestDto;
+import com.example.bookstoreapplication.exception.EntityNotFoundException;
+import com.example.bookstoreapplication.mapper.CategoryMapper;
+import com.example.bookstoreapplication.model.Category;
+import com.example.bookstoreapplication.repository.category.CategoryRepository;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository repository;
+    private final CategoryMapper mapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return repository.findAll().stream().map(mapper::toDto).toList();
```

Comment:
```suggestion
        return repository.findAll().stream().map(mapper::toDto).toList();
```
let's map all categories in the `CategoryMapper`